src/modules/dashboard/components/CoinMachine/BuyTokens/BuyTokens.tsx

```diff
@@ -86,10 +86,11 @@ const TELL_ME_MORE_LINK = '';
 
 type Props = {
   colony: Colony;
+  userHasProfile: boolean;
```

Comment:
The hooks exist to avoid props drilling. I would remove this prop from both components and use useLoggedInUser hook which is already used in BuyTokens component

Reply:
Well, it avoids doing this logic `!!username && !ethereal` several times in child components and it's only one level, but i don't mind either.